extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/common/gphotos/GPhotosUpload.java

```diff
@@ -118,23 +116,29 @@ public GPhotosUpload(
   /**
    * Imports all `items` by fanning out to `batchImporter` upload calls as specified by the.
    *
-   * Returns the number of uploaded bytes, as summed across all `items` that were uploaded.
+   * <p>Returns the number of uploaded bytes, as summed across all `items` that were uploaded.
    */
   // TODO(aksingh737,jzacsh) WARNING: delete the duplicated GooglePhotosImporter code by pulling
   // this out of Media into a new GphotoMedia class that exposes these methods for _both_
   // GoogleMediaImporter _and_ GooglePhotosImporter to use.
   public <T extends DownloadableFile> long uploadItemsViaBatching(
-      Collection<T> items,
-      ItemBatchUploader<T> importer)
-      throws Exception {
+      Collection<T> items, ItemBatchUploader<T> importer) throws Exception {
     long bytes = 0L;
     if (items == null || items.size() <= 0) {
       return bytes;
     }
     Map<String, List<T>> itemsByAlbumId =
         items.stream()
-            .filter(item -> !executor.isKeyCached(item.getIdempotentId()))
+            .filter(
+                item -> !executor.isKeyCached(item.getIdempotentId()) && item.getFolderId() != null)
             .collect(Collectors.groupingBy(DownloadableFile::getFolderId));
+    // Null album-id items get sent here into the empty string key
+    itemsByAlbumId.put(
+        "",
```

Comment:
Only thing left to consider is whether or not an empty string is a possible album id. In the case that it is, this code will overwrite those items. 

Maybe to be safe I can change the upload loop to upload the items in albums and then I can separately loop through the album-less items and upload them, without having to add them to the map.

Reply:
maybe instead of empty string, we could use some other identifier that wont be an album id? something like "$JOB_ID$_ALBUMLESS_ITEMS" ?

Reply:
Done,  made the album-less identifier be the job id appended to "_ALBUMLESS_ITEMS", and tested locally. Thanks Siham!